spot_wrapper/spot_mission_wrapper.py

```diff
@@ -0,0 +1,189 @@
+        self._lease_client = lease_client
+        self._robot_state = robot_state
+        self._spot_check_resp = None
+        self._lease = None
+        self._lease_wallet: LeaseWallet = self._lease_client.lease_wallet
+
+    def _get_lease(self) -> Lease:
+        self._lease = self._lease_wallet.get_lease()
+        return self._lease
+
+    def _load_mission(self, root: nodes_pb2.Node, leases=[], data_chunk_byte_size=None):
+        """Load a mission
+        Args:
+            root: Root node in a mission.
+            leases: All leases necessary to initialize a mission.
+            data_chunk_byte_size: Optional max size of each streamed message
+        Raises:
+            RpcError: Problem communicating with the robot.
+            bosdyn.mission.client.CompilationError: The mission failed to compile.
+            bosdyn.mission.client.ValidationError: The mission failed to validate.
+        """
+        if data_chunk_byte_size:
+            return self._load_mission_as_chunks(root, leases, data_chunk_byte_size)
+        try:
+            resp = self._mission_client.load_mission_async(root, leases)
+        except RpcError:
+            resp = (False, "Could not communicate with the robot")
+        except CompilationError as e:
+            resp = (False, f"The mission failed to compile: {e}")
+        except ValidationError as e:
+            resp = (False, f"The mission could not be validated: {e}")
+        return resp
+
+    def _load_mission_as_chunks(self, root: nodes_pb2.Node, leases=[], data_chunk_byte_size: int = 1000 * 1000):
+        """Load a mission onto the robot.
+        Args:
+            root: Root node in a mission.
+            leases: All leases necessary to initialize a mission.
+            data_chunk_byte_size: max size of each streamed message
+        Raises:
+            RpcError: Problem communicating with the robot.
+            bosdyn.mission.client.CompilationError: The mission failed to compile.
+            bosdyn.mission.client.ValidationError: The mission failed to validate.
+        """
+        try:
+            resp = self._mission_client.load_mission_as_chunks2(root, leases, data_chunk_byte_size)
+        except RpcError:
+            resp = (False, "Could not communicate with the robot")
+        except CompilationError as e:
+            resp = (False, f"The mission failed to compile: {e}")
+        except ValidationError as e:
+            resp = (False, f"The mission could not be validated: {e}")
+        return resp
+
+    def _get_mission_info(self):
+        """Get static information about the loaded mission.
+
+        Raises:
+            RpcError: Problem communicating with the robot.
+        """
+        try:
+            return self._mission_client.get_info()
+        except RpcError:
+            return False, "Could not communicate with the robot"
+
+    def _play_mission(
+        self,
+        pause_time_secs: int,
+        leases=[],
```

Comment:
You shouldn't use a mutable type as a default parameter. You should have the default be `None`, then internal to the function you can check if `None` and set as `[]`